androidApp/src/main/java/dev/johnoreilly/confetti/MainActivity.kt

```diff
@@ -19,18 +22,31 @@ import dev.johnoreilly.confetti.settings.DarkThemeConfig
 import dev.johnoreilly.confetti.settings.SettingsViewModel
 import dev.johnoreilly.confetti.settings.ThemeBrand
 import dev.johnoreilly.confetti.settings.UserEditableSettings
+import dev.johnoreilly.confetti.splash.SplashReadyStatus
 import dev.johnoreilly.confetti.ui.ConfettiApp
 import dev.johnoreilly.confetti.ui.ConfettiTheme
 import dev.johnoreilly.confetti.ui.component.ConfettiBackground
 import kotlinx.coroutines.launch
+import org.koin.android.ext.android.inject
 import org.koin.androidx.viewmodel.ext.android.viewModel
 
 
 class MainActivity : ComponentActivity() {
 
+    private val confettiRepository: ConfettiRepository by inject()
+    private val splashReadyStatus: SplashReadyStatus by inject()
+
     override fun onCreate(savedInstanceState: Bundle?) {
+        val splashScreen = installSplashScreen()
         super.onCreate(savedInstanceState)
 
+        if (intent.data == null) { // If we're deep linking anywhere, don't show splash
```

Comment:
What about doing this based on the nav stack?  If you pull up the nav controller to the activity, then you can probably just check if we are on the home screen.